salt/metalk8s/kubeadm/init/addons/kube-proxy.sls

```diff
@@ -0,0 +1,219 @@
+    - name: kube-proxy
+    - namespace: kube-system
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - requires:
+      - pkg: Install Python Kubernetes client
+
+Deploy kube-proxy (ClusterRoleBinding):
+  kubernetes.clusterrolebinding_present:
+    - name: kubeadm:node-proxier
+    - role_ref:
+        apiGroup: rbac.authorization.k8s.io
+        kind: ClusterRole
+        name: system:node-proxier
+    - subjects:
+      - kind: ServiceAccount
+        name: kube-proxy
+        namespace: kube-system
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+
+    - requires:
+      - pkg: Install Python Kubernetes client
+      - kubernetes: Deploy kube-proxy (ServiceAccount)
+
+Deploy kube-proxy (ConfigMap):
+  kubernetes.configmap_present:
+    - name: kube-proxy
+    - namespace: kube-system
+    - labels:
+        app: kube-proxy
+    - data:
+        config.conf: |-
+          apiVersion: kubeproxy.config.k8s.io/v1alpha1
+          bindAddress: 0.0.0.0
+          clientConnection:
+            acceptContentTypes: ""
+            burst: 10
+            contentType: application/vnd.kubernetes.protobuf
+            kubeconfig: /var/lib/kube-proxy/kubeconfig.conf
+            qps: 5
+          clusterCIDR: {{ networks.pod }}
+          configSyncPeriod: 15m0s
+          conntrack:
+            max: null
+            maxPerCore: 32768
+            min: 131072
+            tcpCloseWaitTimeout: 1h0m0s
+            tcpEstablishedTimeout: 24h0m0s
+          enableProfiling: false
+          healthzBindAddress: 0.0.0.0:10256
+          hostnameOverride: {{ host_name }}
+          iptables:
+            masqueradeAll: false
+            masqueradeBit: 14
+            minSyncPeriod: 0s
+            syncPeriod: 30s
+          ipvs:
+            excludeCIDRs: null
+            minSyncPeriod: 0s
+            scheduler: ""
+            syncPeriod: 30s
+          kind: KubeProxyConfiguration
+          metricsBindAddress: 127.0.0.1:10249
+          mode: ""
+          nodePortAddresses:
+{%- for address in salt['network.ip_addrs'](cidr=networks.workload_plane) %}
+          - {{ address }}/32
+{%- endfor %}
+          oomScoreAdj: -999
+          portRange: ""
+          resourceContainer: /kube-proxy
+          udpIdleTimeout: 250ms
+        kubeconfig.conf: |-
+          apiVersion: v1
+          kind: Config
+          clusters:
+          - cluster:
+              certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+              server: {{ apiserver }}
+            name: default
+          contexts:
+          - context:
+              cluster: default
+              namespace: default
+              user: default
+            name: default
+          current-context: default
+          users:
+          - name: default
+            user:
+              tokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
```

Comment:
Personnally prefer to have data block at the end of the state only have `require` `watch` `onchanges` ... after the big bloc of `data`

Reply:
I don't see the difference between `kubeconfig` or `context` and `require`, `watch`,...? Where would you want them exactly?

Reply:
`kubeconfig` `context` it's `kubernetes` specific stuff `require`, `watch`, ... it's salt orchestration stuff

Reply:
Done